### PR TITLE
[Doc] Add QA labels documentation in `contributing.md`

### PR DESCRIPTION
**Pull request by amenasria**
### What does this PR do?

This PR adds documentation on the `qa/*` labels.

### Motivation

Better doc.

### CODE REVIEW

File: docs/dev/contributing.md [hunk 1]
```diff
@@ -291,6 +291,20 @@ labels that can be use:
 - `need-change/operator`, `need-change/helm`: indicate that the configuration needs to be modified in the operator / helm chart as well.
 - `k8s/<min-version>`: indicate the lowest Kubernetes version compatible with the PR's feature.
 - `backport/<branch-name>`: Add this label to have your changes automatically backported to `<branch-name>`.
+- `qa/done` or `qa/no-code-change`: used to skip the QA week:
+  - `qa/done` label is recommended in case of code changes **and** manual / automated qa done before merge.
+  - `qa/no-code-change` is recommended if there's no code changes in the Agent binary code.
+
+> [!NOTE]
+> Use `qa/no-code-change` if your PR only changes tests or a module/package that does not end up in the agent build
+> All of the follows require QA, so cannot use `qa/no-code-change`:
```
Comment (chouetz) [inline docs/dev/contributing.md:300]:
```suggestion
> All of the following require QA, so cannot use `qa/no-code-change`:
```
I'm not sure at all but I guess I'm more used to seeing this one.

**Review by chouetz — APPROVED** [review]

File: docs/dev/contributing.md (hunk 1 above)
Comment (amenasria) [reply docs/dev/contributing.md:300]: image
Google agrees with you, fixing, thanks !

**Review by urseberry — APPROVED** [review]
The Documentation team does not have an opinion on these labels, approving.

File: docs/dev/contributing.md [hunk 2]
```diff
@@ -291,6 +291,20 @@ labels that can be use:
...
+> [!NOTE]
+> Use `qa/no-code-change` if your PR only changes tests or a module/package that does not end up in the agent build
+> All of the following require QA, so cannot use `qa/no-code-change`:
+> - New feature
+> - Fixing a typo in a log message
+> - Changing the log level
+> - Renaming a function
+> - Removing code
+> - Creating a module and update go import
+> - Dependency bump
```
[the remark this answers is not in the document]
Comment (amenasria) [reply docs/dev/contributing.md:307]: Good point, adding this

**Pull request merged**